src/components/scroll-up-button/ScrollUpButton.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react"
+
+import Button from "../button/Button"
+import { L_DOWN } from "../../utils/viewport"
+import { SCROLL_UP } from "./ScrollUpButton.constants"
+import styled, { keyframes } from "styled-components"
+import theme from "../../utils/theme"
+
+const show = keyframes`
+  from {
+    transform: scale(0);
+  }
+
+  to {
+    transform: scale(1);
+  }
+`
+
+const Container = styled.div`
+  position: fixed;
+  bottom: 20px;
+  right: 20px;
+  background-color: ${theme.black};
+
+  @media ${L_DOWN} {
+    left: 20px;
+    right: auto;
+  }
+
+  .scrollUp-show {
+    animation: ${show} 0.4s ease-in-out 0s forwards;
+  }
+
+  .scrollUp-hide {
+    display: none;
+  }
+`
+
+export const ScrollUpButton = () => {
+  const [currentScrollPosition, setCurrentScrollPosition] = useState(0)
+  const { stepUpValue, timeInMiliseconds, visibilityLimit } = SCROLL_UP
+
+  window.addEventListener("scroll", () => {
+    setCurrentScrollPosition(document.documentElement.scrollTop)
+  })
+
+  const handleScrollUp = () => {
```

Comment:
I think we need to create hook for that inside our utils directory -> it may be re-used in future and also will make any code cleaner so:

```ts
const useScrollTo= () => {
   const scrollTop= () => {

   }

   const scrollBottom = () => {

   }

   return { scrollTop, scrollBottom }
}
```

and use it inside component.